docs/how-to/debugging-guide.rst

```diff
@@ -0,0 +1,245 @@
+
+   .. code:: shell
+
+      mpirun -np <numProcesses> ./rcclReplayer </path/to/logfile> <numGpusPerMpiRank>
+
+   In a multi-node application environment, you can replay the collective logs on multiple nodes
+   using the following command:
+
+   .. code:: shell
+
+      mpirun --hostfile <path/to/hostfile.txt> -np <numProcesses> ./rcclReplayer </path/to/logfile> <numGpusPerMpiRank>
+
+   .. note::
+
+      Depending on the MPI library you are using, you might need to modify the ``mpirun`` command.
+
+Troubleshooting
+=============================
+
+Use the following troubleshooting techniques to potentially isolate the issue.
+
+*  Build or run the develop branch version of RCCL and see if the problem persists.
+*  Try an earlier minor or major releases of RCCL.
+*  If you recently made changes to the ROCm runtime configuration, KFD/driver or compiler,
+   run the test again with the previous configuration.
+
+.. _analyze-performance-info:
+
+Analyzing performance issues
+=============================
+
+If the issues involve performance issues and occur in an e2e workload, try the following 
+microbenchmarks and collect the results. Follow the instructions in the subsequent sections
+to run these benchmarks and provide the results to the support team.
+
+*  TransferBench
+*  RCCL Unit Tests
+*  rccl-tests
+  
+Collect the TransferBench data
+---------------------------------
+
+TransferBench allows you to benchmark simulataneous copies between
+user-specified devices. For more information, 
+see :doc:`the TransferBench documentation <transferbench:index>`.
+
+To collect the TransferBench data, follow these steps:
+
+#. Clone the TransferBench Git repository.
+
+   .. code:: shell
+
+      git clone https://github.com/ROCm/TransferBench.git 
+
+#. Change to the new directory and build the component.
+
+   .. code:: shell
+
+      cd TransferBench
+      make
+
+#. Run the TransferBench utlity with the following parameters and save the results.
+
+   .. code:: shell
+
+      USE_FINE_GRAIN=1 GFX_UNROLL=2 ./TransferBench a2a 64M 8
+
+Collect the RCCL microbenchmark data
+-------------------------------------
+
+To use the RCCL tests to collect the RCCL benchmark data, follow these steps:
+
+#. Disable NUMA auto-balancing using the following command:
+
+   .. code:: shell
+
+      sudo sysctl kernel.numa_balancing=0
+
+   Run the following command to verify the setting. The expected output is ``0``.
+
+   .. code:: shell
+
+      cat /proc/sys/kernel/numa_balancing
+
+#. Build MPI, RCCL, and rccl-tests. To download and install MPI, see the `MPI site <https://www.mpich.org/>`_.
+   To learn how to build and run rccl-tests, see the `rccl-tests GitHub <https://github.com/ROCm/rccl-tests/blob/develop/README.md>`_.
+
+#. Run rccl-tests with MPI and collect the performance numbers.
+
+RCCL and NCCL comparisons
+=============================
+
+If you are also using NVIDIA HW or NCCL and notice a performance gap between the two systems,
```

Comment:
IS HW supposed to be hardware, or is it actually something?

Reply:
It's HW. I'll spell it out.